### PR TITLE
chore: Add Github Actions CI

### PR DESCRIPTION
**Pull request by andreubotella**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - main
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
```
Comment (MierenManz) [inline .github/workflows/ci.yml:18]: Maybe seperate the test step and only have the timeout there? Just a suggestion. This seems fine for now aswell

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    branches:
+      - main
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
```
Comment (MierenManz) [inline .github/workflows/ci.yml:17]: Could you add a todo so that we run ci on all 3 platforms rather than Linux only?

**Review by MierenManz — APPROVED** [review]
Other than the suggestion and comment
Lgtm

File: .github/workflows/ci.yml (hunk 1 above)
Comment (andreubotella) [reply .github/workflows/ci.yml]: I've set the build's timeout to 40 minutes and the test step's to 20. Is that okay?

**Comment by MierenManz** [comment]
Lgtm!

**Pull request merged**